clients/drcachesim/analyzer.h

```diff
@@ -125,7 +125,7 @@ template <typename RecordType, typename ReaderType> class analyzer_tmpl_t {
         analyzer_tool_shard_data_t(analyzer_tool_shard_data_t &&src)
         {
             shard_data = src.shard_data;
-            interval_snapshot_data = std::move(interval_snapshot_data);
+            interval_snapshot_data = std::move(src.interval_snapshot_data);
```

Comment:
What were the consequences?  Do we need to add a unit test case that would catch this?

Reply:
@ksco mentioned that the LicheePi4A (RISC-V SBC) compiler complained about this. I don't think this actually crashed anything.

Reply:
https://github.com/DynamoRIO/dynamorio/pull/6022#discussion_r1197255790

Reply:
Created #6060 to improve our compiler warnings.

Reply:
But what are the actual consequences of having this bug?  What code calls this function?  What happens without doing the requested move?

Reply:
If you're saying there are no consequences and all you're worried about is a compiler warning, then let's delete this function.

Reply:
We need the move constructor for the same reason we have it for analyzer_worker_data_t: for it to work with std::vector.

The bug fixed in this PR could've been problematic if the `analyzer_shard_data_t::tool_data`  vector did an internal re-allocation (in response to addition of new elements). But we allocate enough space in the vector for all tools using an upfront `resize(num_tools_)`, so it wouldn't cause any real issues I think.

The move constructor for analyzer_shard_data_t though could be deleted. The current code uses an unordered_map for storing analyzer_shard_data_t, not a vector.

I sent PR #6066 for this cleanup and some other misc stuff that should've been done before.